dandi/validate.py

```diff
@@ -1,19 +1,59 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import Enum
 from pathlib import Path
 from typing import Iterator, List, Optional, Tuple, Union
 
 import appdirs
 
 from .files import find_dandi_files
 
-# TODO: provide our own "errors" records, which would also include warnings etc
+BIDS_TO_DANDI = {
+    "subject": "subject_id",
+    "session": "session_id",
+}
+
+
+@dataclass
+class ValidationResult:
+    id: str
+    origin: ValidationOrigin
+    scope: Scope
+    asset_paths: Optional[list[str]] = None
+    dandiset_path: Optional[Path] = None
+    dataset_path: Optional[Path] = None
+    message: str = ""
```

Comment:
Remove the default value and move the field above `asset_paths`.

Reply:
Message is only defined for errors, not for matches, so it needs to have a default value, or am I missing something?
We could put some generic placeholder message but “File identified” over and over is just wasted memory.

Reply:
Then it would probably be better to change the type to `Optional[str]` and set the default to `None` for representing message-less results.